From 95e9646cc05dc6b2f3caa197797e1bc0a381c406 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 25 Jun 2025 19:38:00 +0200
Subject: [PATCH 0997/1010] evl/net: socket: drop get_netif handler

We have no more users of this handler, which never made much sense
anyway, since we use the binding information locally in the protocol
layers where we always know where to find it. IOW, there should not be
cross-layer requests for binding information which would require such
abstraction/indirection.

Drop this cruft.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/evl/net/socket.h       |  1 -
 kernel/evl/net/ipv4/udp.c      | 13 -------------
 kernel/evl/net/packet/packet.c |  9 ++++-----
 3 files changed, 4 insertions(+), 19 deletions(-)

diff --git a/include/evl/net/socket.h b/include/evl/net/socket.h
index 1fcc9895cf..e715c55869 100644
--- a/include/evl/net/socket.h
+++ b/include/evl/net/socket.h
@@ -54,7 +54,6 @@ struct evl_net_proto {
 			size_t iovlen);
 	__poll_t (*oob_poll)(struct evl_socket *esk,
 			struct oob_poll_wait *wait);
-	struct net_device *(*get_netif)(struct evl_socket *esk);
 	void (*handle_offload)(struct evl_socket *esk);
 };
 
diff --git a/kernel/evl/net/ipv4/udp.c b/kernel/evl/net/ipv4/udp.c
index ad1758b67d..c795a2ab20 100644
--- a/kernel/evl/net/ipv4/udp.c
+++ b/kernel/evl/net/ipv4/udp.c
@@ -648,18 +648,6 @@ static __poll_t poll_udp(struct evl_socket *esk,
 	return 0;
 }
 
-/* in-band */
-static struct net_device *get_netif_udp(struct evl_socket *esk)
-{
-	int ifindex;
-
-	ifindex = READ_ONCE(esk->sk->sk_bound_dev_if);
-	if (ifindex)
-		return evl_net_get_dev_by_index(esk->net, ifindex);
-
-	return NULL;
-}
-
 /* in-band */
 static void handle_udp_inband(struct evl_socket *esk)
 {
@@ -911,6 +899,5 @@ struct evl_net_proto evl_net_udp_proto = {
 	.oob_send = send_udp,
 	.oob_poll = poll_udp,
 	.oob_receive = receive_udp,
-	.get_netif = get_netif_udp,
 	.handle_offload = handle_udp_inband,
 };
diff --git a/kernel/evl/net/packet/packet.c b/kernel/evl/net/packet/packet.c
index f42585069a..7504bd042d 100644
--- a/kernel/evl/net/packet/packet.c
+++ b/kernel/evl/net/packet/packet.c
@@ -356,7 +356,7 @@ static int bind_packet_socket(struct evl_socket *esk,
 	return ret;
 }
 
-static struct net_device *get_netif_packet(struct evl_socket *esk)
+static struct net_device *get_netif(struct evl_socket *esk)
 {
 	return  evl_net_get_dev_by_index(esk->net,
 					esk->u.packet.ifindex);
@@ -384,7 +384,7 @@ static struct net_device *find_xmit_device(struct evl_socket *esk,
 			if (namelen)
 				return ERR_PTR(-EINVAL);
 
-			dev = get_netif_packet(esk);
+			dev = get_netif(esk);
 		} else {
 			if (namelen < sizeof(addr))
 				return ERR_PTR(-EINVAL);
@@ -401,7 +401,7 @@ static struct net_device *find_xmit_device(struct evl_socket *esk,
 			dev = evl_net_get_dev_by_index(esk->net, addr.sll_ifindex);
 		}
 	} else {
-		dev = get_netif_packet(esk);
+		dev = get_netif(esk);
 	}
 
 	if (dev == NULL)
@@ -667,7 +667,7 @@ static __poll_t poll_packet(struct evl_socket *esk,
 	if (!list_empty(&esk->input) || evl_test_socket_iots(esk))
 		ret = POLLIN|POLLRDNORM;
 
-	dev = get_netif_packet(esk);
+	dev = get_netif(esk);
 	if (dev) {
 		est = dev->oob_state.estate;
 		evl_poll_watch(&est->poll_head, wait, NULL);
@@ -686,7 +686,6 @@ static struct evl_net_proto ether_packet_proto = {
 	.oob_send = send_packet,
 	.oob_poll = poll_packet,
 	.oob_receive = receive_packet,
-	.get_netif = get_netif_packet,
 };
 
 static struct evl_net_proto *
-- 
2.30.2

